src/herbpy/action/blocks.py

```diff
@@ -0,0 +1,172 @@
+            # Move down until touching the table
+            with env:
+                start_point = manip.GetEndEffectorTransform()[0:3, 3]
+                table_aabb = ComputeEnabledAABB(table)
+                table_height = table_aabb.pos()[2] + table_aabb.extents()[2]
+                # 0.14 is the distance from finger tip to end-effector frame
+                current_finger_height = manip.GetEndEffectorTransform()[2,3] - 0.14
+                min_distance = 0.10# current_finger_height - table_height
+
+            down_direction = [0., 0., -1.]
+            with RenderVector(start_point, down_direction, min_distance, env):
+                manip.MoveUntilTouch(direction=down_direction, timelimit=5,
+                    distance=min_distance, max_distance=min_distance + 0.05,
+                    ignore_collisions=blocks + [table])
+
+            # Move parallel to the table to funnel the block into the fingers
+            # by projecting the -x direction of end-effector onto the xy-plane.
+            with env:
+                start_point = manip.GetEndEffectorTransform()[0:3, 3]
+                funnel_direction = -1. * manip.GetEndEffectorTransform()[:3,0]
+                funnel_direction[2] = 0.
+
+            # TODO: We should only have to disable the block for this. Why does
+            # this fail if we do not disable the table?
+            with AllDisabled(env, blocks + [table]):
+                with RenderVector(start_point, funnel_direction, 0.1, env):
+                    manip.PlanToEndEffectorOffset(direction=funnel_direction,
+                                                  distance=0.08, max_distance=0.12, 
+                                                  timelimit=5., execute=True)
+        
+        # Close the finger to grab the block
+        manip.hand.MoveHand(f3=1.7)
+            
+        # Compute the pose of the block in the hand frame
+        with env:
+            local_p = [0.01, 0, 0.24, 1.0]
+            hand_pose = manip.GetEndEffectorTransform()
+            world_p = numpy.dot(hand_pose, local_p)
+            block_pose = block.GetTransform()
+            block_pose[:,3] = world_p
+            block_relative = numpy.dot(numpy.linalg.inv(hand_pose), block_pose)
+        
+        # Now lift the block up off the table
+        with AllDisabled(env, blocks + [table]):
+            manip.PlanToEndEffectorOffset(direction=[0, 0, 1], distance=0.05, 
+                                          timelimit=5, execute=True)
+            
+        # OpenRAVE trick to hallucinate the block into the correct pose relative to the hand
+        with env:
+            hand_pose = manip.GetEndEffectorTransform()
+            block_pose = numpy.dot(hand_pose, block_relative)
+            block.SetTransform(block_pose)
+            manip.GetRobot().Grab(block)
+
+    except PlanningError as e:
+        logger.error('Failed to complete block grasp')
+        raise
+    finally:
+        return block
```

Comment:
I don't think it makes sense to put this `return` in the `finally` block. I would remove the `finally` block entirely and put it in the function scope.